haystack/management/commands/update_index.py

```diff
@@ -85,6 +85,8 @@ def do_update(backend, index, qs, start, end, total, verbosity=1, commit=True):
     # in memory. Useful when reindexing large amounts of data.
     small_cache_qs = qs.all()
     current_qs = small_cache_qs[start:end]
+    
+    index.pre_process_data(current_qs)
```

Comment:
Looking at this API more, I'm inclined to say this should be documented as `current_qs = index.preprocess_queryset(current_qs)` so you could potentially call things like `select_related`, `prefetch_related`, etc. or even replace the queryset entirely if there's a good reason.